pennylane/transforms/zx/converter.py

```diff
@@ -0,0 +1,587 @@
+
+            # The graph is not diagram like.
+            if len(neighbors) != 1:
+                raise qml.QuantumFunctionError(
+                    "Graph doesn't seem circuit like: multiple parents. Try to use the PyZX function `extract_circuit`."
+                )
+
+            neighbor_0 = neighbors[0]
+
+            if qubits[neighbor_0] != qubit_1:
+                raise qml.QuantumFunctionError(
+                    "Cross qubit connections, the graph is not circuit-like."
+                )
+
+            # Add Hadamard gate (written in the edge)
+            if graph.edge_type(graph.edge(neighbor_0, vertex)) == EdgeType.HADAMARD:
+                operations.append(qml.Hadamard(wires=qubit_1))
+
+            # Vertex is a boundary
+            if type_1 == VertexType.BOUNDARY:
+                continue
+
+            # Add the one qubits gate
+            operations.extend(_add_one_qubit_gate(param, type_1, qubit_1, decompose_phases))
+
+            # Given the neighbors on the same rowadd two qubits gates
+            neighbors = [
+                w for w in graph.neighbors(vertex) if graph_rows[w] == row_key and w < vertex
+            ]
+
+            for neighbor in neighbors:
+                type_2 = types[neighbor]
+                qubit_2 = qubits[neighbor]
+
+                operations.extend(
+                    _add_two_qubit_gates(graph, vertex, neighbor, type_1, type_2, qubit_1, qubit_2)
+                )
+
+    qscript = QuantumScript(operations)
+    return qscript
+
+
+def _add_one_qubit_gate(param, type_1, qubit_1, decompose_phases):
+    """Return the list of one qubit gates, that will be added to the tape."""
+    if decompose_phases:
+        type_z = type_1 == VertexType.Z
+        if type_z and param.denominator == 2:
+            op = qml.adjoint(qml.S(wires=qubit_1)) if param.numerator == 3 else qml.S(wires=qubit_1)
+            return [op]
+        if type_z and param.denominator == 4:
+            if param.numerator in (1, 7):
+                op = (
+                    qml.adjoint(qml.T(wires=qubit_1))
+                    if param.numerator == 7
+                    else qml.T(wires=qubit_1)
+                )
+                return [op]
+            if param.numerator in (3, 5):
+                op1 = qml.PauliZ(wires=qubit_1)
+                op2 = (
+                    qml.adjoint(qml.T(wires=qubit_1))
+                    if param.numerator == 3
+                    else qml.T(wires=qubit_1)
+                )
+                return [op1, op2]
+        if param == 1:
+            op = qml.PauliZ(wires=qubit_1) if type_1 == VertexType.Z else qml.PauliX(wires=qubit_1)
+            return [op]
+        if param != 0:
+            scaled_param = np.pi * float(param)
+            op_class = qml.RZ if type_1 == VertexType.Z else qml.RX
+            return [op_class(scaled_param, wires=qubit_1)]
+    # Phases are not decomposed
+    elif param != 0:
```

Comment:
```suggestion
    # Phases are not decomposed
    if param != 0:
```
Pretty sure this would work.